src/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -63,6 +63,26 @@ public static void ReadClassWithComplexObjects()
             obj.Verify();
         }
 
+        [Theory]
+        [InlineData("", " ")]
+        [InlineData("", "\t ")]
+        [InlineData("", "//Single Line Comment\r\n")]
+        [InlineData("", "/* Multi\nLine Comment */")]
+        [InlineData("", "\t\t\t\n// Both\n/* Comments */")]
+        [InlineData(" ", "")]
+        [InlineData("\t ", "")]
+        [InlineData("// Leading Comment\n", "")]
+        [InlineData("/* Multi\nLine\nComment */ ", "")]
+        [InlineData("/* Multi\nLine\nComment */ ", "\t// trailing comment\n ")]
+        public static void ReadClassIgnoresLeadingOrTrailingTrivia(string leadingTrivia, string trailingTrivia)
+        {
+            var options = new JsonSerializerOptions();
+            options.ReadCommentHandling = JsonCommentHandling.Skip;
+
+            ClassWithComplexObjects obj = JsonSerializer.Parse<ClassWithComplexObjects>(leadingTrivia + ClassWithComplexObjects.s_json + trailingTrivia, options);
```

Comment:
As an aside (unrelated to this PR), we should consider adding tests where we have comments in-between the JSON objects as well to make sure the serializer continues to handle that. We have such tests for the reader/document, but I am not sure we have them for the serializer.